### PR TITLE
Add sgn kernel & sign kernel & sgn op

### PR DESCRIPTION
**Pull request by AlienLiang23**
Add sgn kernel & sign kernel & sgn op

### CODE REVIEW

**Comment by AlienLiang23** [comment]
image
all cases passed except **TestCommonXPU.test_dtypes_sgn_xpu** which is already in the skip list

File: src/aten/sycl/UnaryKernels.cpp [hunk 1]
```diff
@@ -266,4 +292,23 @@ void bitwise_not_kernel(TensorIteratorBase& iter) {
   }
 }
 
+void sgn_kernel(TensorIteratorBase& iter) {
+  AT_DISPATCH_COMPLEX_TYPES_AND(kComplexHalf, iter.dtype(), "sgn_xpu", [&] {
+    using opmath_t = at::opmath_type<scalar_t>;
+    gpu_kernel(iter, SgnFunctor<opmath_t>());
```
Comment (fengyuan14) [inline src/aten/sycl/UnaryKernels.cpp:298]:
```suggestion
    gpu_kernel(iter, SgnFunctor<scalar_t>());
```

File: src/aten/sycl/UnaryKernels.cpp [hunk 2]
```diff
@@ -266,4 +292,23 @@ void bitwise_not_kernel(TensorIteratorBase& iter) {
   }
 }
 
+void sgn_kernel(TensorIteratorBase& iter) {
+  AT_DISPATCH_COMPLEX_TYPES_AND(kComplexHalf, iter.dtype(), "sgn_xpu", [&] {
+    using opmath_t = at::opmath_type<scalar_t>;
```
Comment (fengyuan14) [inline src/aten/sycl/UnaryKernels.cpp:297]:
```suggestion
```

File: src/aten/sycl/UnaryKernels.cpp [hunk 3]
```diff
@@ -77,6 +77,32 @@ struct NegFunctor {
...
+struct SgnFunctor {
+  scalar_t operator()(scalar_t a) const {
+    using opmath_t = at::opmath_type<scalar_t>;
+    if (a == scalar_t(0, 0)) {
+      return scalar_t(0, 0);
+    } else {
+      return a / std::abs(static_cast<opmath_t>(a));
+    }
+  }
+};
```
Comment (fengyuan14) [inline src/aten/sycl/UnaryKernels.cpp:90]:
```suggestion
struct SgnFunctor {
  scalar_t operator()(scalar_t z_) const {
    using opmath_t = at::opmath_type<scalar_t>;
    opmath_t z = static_cast<opmath_t>(z_);
    if (z == opmath_t(0, 0)) {
      return opmath_t(0, 0);
    } else {
      return z / std::abs(z);
    }
  }
};
```

**Review by fengyuan14 — APPROVED** [review]

**Pull request merged**